3174 main page banner fixes

therocket-gr opened the pull request:
Related to #
3174-gh-tumblr/a8c-dsp

## Proposed Changes

We want to change the logic of this banner so it never gets completely hidden..

the user can collapse it and expand it. but it will show a collapsed version instead of getting completely hidden


the result looks like the below

1 when expanded (this is the normal state for the 1st time)
![Screenshot 2025-02-20 at 1 25 47 PM](https://github.com/user-attachments/assets/62005fad-7e14-439f-b368-fa8f414ef62c)

2 when collapsed
![Screenshot 2025-02-20 at 1 25 41 PM](https://github.com/user-attachments/assets/3d69ebe7-5b10-4242-a298-7defaa463d09)

## Why are these changes being made?
i changed the logic (and naming of params) to accommodate the change and make it more readable in the code

the new cookie is named
`blaze-tsp-banner-collapsed`

## Testing Instructions

checkout the branch and test locally (you can also test in the preview link i guess)
it would be nice to test with a blog that has less than 3 campaign - so you can test that the tsp banner is taking prioroty over the normal banner

open the promoted posts page (posts or campaigns)

the banner should be present and expanded

click to collapse.. the banner should look collapsed and a new banner should be writen in your browser (named blaze-tsp-banner-collapsed and set to true)

click to expand again.. the cookie should be false

... 

with both states collapsed or not try to refresh the browser.. the banner state should follow your cookie value

some notes 
1. as i see it now a user will never get the normal banner when tsp is eligible .. as we give priority over the tsp banner.. and we don t hide it.. everytime the user will see this one and the normal banner will never be displayed
2. i dont like the fact that we are hiding the subtitle of the page (it is shown above in the ss) because now the page jumps when we collapse and expand the banner..  - i think we should always show it when tsp banner is present and hide it only in the case with the  normal banner



as it is now this PR should be merged after the TSP is released to the public. i removed existing todos and everything should work. but if we merge it now the users will start seeing this new banner


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

therocket-gr committed:
fixed invalid svg params

therocket-gr committed:
fixed styles

therocket-gr committed:
fixed banner component

therocket-gr committed:
fixed main component logic to accommodate the change in banner

timur987 committed:
Try fixing an e2e test for WP Blaze

timur987 reviewed (approved):
Thanks @therocket-gr!

The change LGTM and works as expected.
I noticed that E2E tests failed due to missing the regular banner. This commit (202c68bb2a4e17816e5acd1784c1969a2c1e69d0) fixes it.

We discussed that this PR should not be delivered until the TSP feature release, but I don't think we should wait.

What would you say about applying this patch instead: [patch.txt](https://github.com/user-attachments/files/18887192/patch.txt) ?

What is does is completely forbid TSP banners, so for release we will need to just a couple of lines of code.

sbarbosa reviewed (approved):
LGTM 👍

I Agree with Timur. We can hardcode a false flag and merge the PR. We already have ticket DSP 3153 that will remove of these hardcoded flags and open the widget for all users.

If you can, include a comment on that ticket mentioning the location of the flag, so its easier to remove when we open this.

therocket-gr commented:
@timur987 and @sbarbosa please also comment on those notes

some notes

as i see it now a user will never get the normal banner when tsp is eligible .. as we give priority over the tsp banner.. and we don t hide it.. everytime the user will see this one and the normal banner will never be displayed
i dont like the fact that we are hiding the subtitle of the page (it is shown above in the ss) because now the page jumps when we collapse and expand the banner.. - i think we should always show it when tsp banner is present and hide it only in the case with the normal banner

timur987 commented:
Thanks for starting a discussion @therocket-gr!

Indeed, it jumps:

![tsp-banner-collapse-expand](https://github.com/user-attachments/assets/47701748-5c10-4c5f-bf48-805d81376524)

I bet @bartoszbak can help us here.

therocket-gr commented:
> I would remove a toggler to remove the possibility of page jumping, but let Bart share your vision first.

i think the simplest solution would be to just not remove the subtitle 
my understanding is that we removed it because the normal blaze banner has the same wording the "use blaze to grow..."

now for the tsp banner it has a totally different message.. so i guess it is ok to always keep it there
@timur987

therocket-gr committed:
added the learn more link

therocket-gr committed:
minor fixes and make the banner not show for now
added a todo and will reference the line we need to delete in the relevant ticket

therocket-gr committed:
minor fixes in display logic

therocket-gr commented:
> > I would remove a toggler to remove the possibility of page jumping, but let Bart share your vision first.
> 
> i think the simplest solution would be to just not remove the subtitle my understanding is that we removed it because the normal blaze banner has the same wording the "use blaze to grow..."
> 
> now for the tsp banner it has a totally different message.. so i guess it is ok to always keep it there @timur987
> 
> we can do so by removing the istspcollapsed in this condition
> 
> https://github.com/Automattic/wp-calypso/pull/100067/files#diff-af45fef67dabb389ee3b79b06fd228c64fa0c2e369993c26b3e105c2b1e2cc04R249-R252

i took the liberty to just add the subtitle to be always visible @timur987

bartoszbak commented:
I think the easiest way will be to maintain the subheader as it is in the design

![CleanShot 2025-02-21 at 11 51 09@2x](https://github.com/user-attachments/assets/3ebc113b-fc15-4ee5-bf1b-de946ce136ee)

[Figma link](https://www.figma.com/design/C8gRy5FoZtYxuF2WlvqVSs/WP-Blaze---Super-native-ad-format?node-id=139-9680&t=83uZrmJwDwsWPKfy-1)

timur987 reviewed (approved):
> i took the liberty to just add the subtitle to be always visible @timur987

Thanks for this @therocket-gr!

It looks and works perfectly now 🚀

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17266714

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @therocket-gr for including a screenshot in the description! This is really helpful for our translators.
